Runtime/Reporter/ReportUploadGuardService.cs

```diff
@@ -0,0 +1,37 @@
+using BugSplatUnity.Runtime.Settings;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+namespace BugSplatUnity.Runtime.Reporter
+{
+    internal interface IReportUploadGuardService
+    {
+        public bool ShouldPostLogMessage(IClientSettingsRepository clientSettingsRepository, Exception exception, LogType type);
+        public bool ShouldPostException(IClientSettingsRepository clientSettingsRepository, Exception exception);
+    }
+
+    internal class ReportUploadGuardService : IReportUploadGuardService
+    {
+        public bool ShouldPostException(IClientSettingsRepository clientSettingsRepository, Exception exception)
```

Comment:
since both methods on this class require an IClientSettingsRepository param can we move this to the constructor?

Reply:
Great suggestion! I moved it to a constructor and it really cleaned up the ReportGuardServiceParameters.